qcodes/math/field_vector.py

```diff
@@ -0,0 +1,245 @@
+        :param theta: float, the angle between the vector and the positive z-axis
+        :param phi: float, the angle of the projection of the vector on the xy plane and the positive x-axis
+        :param rho: float, the magnitude of the projection of the vector on the xy plane
+
+        All inputs are optional, however the user needs to either give (x, y, z) values,
+        (r, theta, phi) values or (phi, rho, z) values.
+        """
+        self._x = x
+        self._y = y
+        self._z = z
+
+        self._r = r
+        if theta is not None:
+            self._theta = np.radians(theta)
+        else:
+            self._theta = theta
+
+        if phi is not None:
+            self._phi = np.radians(phi)
+        else:
+            self._phi = phi
+
+        self._rho = rho
+
+        self._compute_unknowns()
+
+    def _set_attribute_value(self, attr_name, value):
+
+        if value is None:
+            return
+
+        attr_value = getattr(self, "_" + attr_name)
+
+        if attr_value is None:
+            setattr(self, "_" + attr_name, value)
+        else:
+            if not np.isclose(attr_value, value):
+                raise ValueError("Computed value of {} inconsistent with given value".format(attr_name))
+
+    def _set_attribute_values(self, attr_names, values):
+
+        for attr_name, value in zip(attr_names, values):
+            self._set_attribute_value(attr_name, value)
+
+    @staticmethod
+    def _cartesian_to_other(x, y, z):
+        """ Convert a cartesian set of coordinates to values of interest."""
+
+        if any([i is None for i in [x, y, z]]):
+            return None
+
+        phi = np.arctan2(y, x)
+        rho = np.sqrt(x ** 2 + y ** 2)
+        r = np.sqrt(x ** 2 + y ** 2 + z ** 2)
+        if r != 0:
+            theta = np.arccos(z / r)
+        else:
+            theta = 0
+
+        return x, y, z, r, theta, phi, rho
+
+    @staticmethod
+    def _spherical_to_other(r, theta, phi):
+        """Convert from spherical to other representations"""
+
+        if any([i is None for i in [r, theta, phi]]):
+            return None
+
+        z = r * np.cos(theta)
+        x = r * np.sin(theta) * np.cos(phi)
+        y = r * np.sin(theta) * np.sin(phi)
+        rho = np.sqrt(x ** 2 + y ** 2)
+
+        return x, y, z, r, theta, phi, rho
+
+    @staticmethod
+    def _cylindrical_to_other(phi, rho, z):
+        """Convert from cylindrical to other representations"""
+
+        if any([i is None for i in [phi, rho, z]]):
+            return None
+
+        x = rho * np.cos(phi)
+        y = rho * np.sin(phi)
+        r = np.sqrt(rho ** 2 + z ** 2)
+        theta = np.arccos(z / r)
+
+        return x, y, z, r, theta, phi, rho
+
+    def _compute_unknowns(self):
+        """
+        Compute all values. This function will raise an error if:
+         1) There are contradictory inputs (e.g. x=3, y=4, z=0 and rho=6)
+        """
+
+        for _ in range(1):
```

Comment:
Why is this for loop needed? 

I would remove it and replace the breaks with return. That should be equivalent?

Reply:
I think a break statement only works in loops.Your suggestions would indeed be equivalent. Ideally we want to have a single exit point in fuctions, however my solution is not ideal either. I will replace the breaks by return.

Reply:
Btw... the situation would be a little trickier if after the loop we had other code. We could not simply return as the rest of the code would not be reachable. A better solution has been implemented in the code. See the updated PR shortly